Meta/gn/secondary/Userland/Libraries/LibTest/BUILD.gn

```diff
@@ -0,0 +1,52 @@
+import("//Meta/gn/build/library_name.gni")
+
+shared_library("LibTest") {
+  output_name = output_name_prefix + "test"
+  include_dirs = [
+    "//Userland/Libraries",
+  ]
+  sources = [
+    "CrashTest.cpp",
+    "CrashTest.h",
+    "Macros.h",
+    "Results.h",
+    "TestCase.h",
+    "TestRunner.h",
+    "TestRunnerUtil.h",
+    "TestSuite.cpp",
+    "TestSuite.h",
+  ]
+
+  deps = [ "//AK" ]
+  if (current_os != "serenity") {
+    deps += [ "//Userland/Libraries/LibCore" ]
+  }
+
+  output_name = output_name_prefix + "test"
+
+  testonly = true
+}
+
+# FIXME: These are OBJECT libraries in the CMake. Is source_set() the correct choice?
```

Comment:
Does gn deduplicate source_sets with the same flags? i.e. is `source_set` the right translation from object library here? or should they be static libs? the fact that it's got main() in it makes me a bit cautious.

Reply:
For a source_set, gn compiles all the source files in it to .o files and then adds all the .o files in it to the first linkable thing that (transitively) depends on it. I think that's what OBJECT libraries do, too. I translate OBJECT libs to source_sets.

But I don't understand your "deduplicate" question.

(In particular, this means all .o files in a source_set are always linked in. A static library on the other hand produce a .a file, which has a symbol->.o file map at the start, and only the .o files in it that are needed by unresolved symbols at link time are pulled in by the linker.)

Reply:
The de-duplicate question is more about thinking that maybe a `source_set` would simply add the files to `sources` of every target that depends on them and then compile the sources once for each each target. Given your explanation above, it sounds like they are smarter than that and do behave like source_sets. so I'll remove that fixme question